src/model/parseModelDeclaration.ts

```diff
@@ -67,12 +69,18 @@ export function parseModelDeclaration<
       }
 
       const relationDefinition = declaration[key] as RelationDefinition<
-        RelationKind.OneOf,
+        any,
         ModelName
       >
-
-      if (exactValue) {
-        if (Array.isArray(exactValue)) {
+      if (exactValue && relationDefinition) {
+        const relationPrimarykey = findPrimaryKey(
```

Comment:
We can assume that primary key exists because we have already checked it when the factory is first created

Reply:
That's a good point. I'd rather parse `primaryKey` in `parseModelDeclaration` and throw away `findPrimaryKey` function. It's confusing why the model that's known is parsed only when an entity is created. Needless to say, that creating new entities doesn't change model declaration's properties or relations. Looks like a design overhaul to me, and I'd like to address it in the follow-up pull request.